pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -989,3 +947,42 @@ func FindPort(pod *v1.Pod, svcPort *v1.ServicePort) (int, error) {
 func createUID(podName, namespace string) string {
 	return "kubernetes://" + podName + "." + namespace
 }
+
+// first phase of instantiating IstioEndpoint, applies pod metadata
+// Note: must be followed by `applyAddressToIstioEndpoint` to build a complete IstioEndpoint
+func (c *Controller) initIstioEndpoint(pod *v1.Pod) model.IstioEndpoint {
```

Comment:
This is definitely better compared to previous approach, What do you think about returning new struct `endPointBuilder` here and `newIstioEndpoint` takes the builder? That way we do not have the caveats of partial buidling?

Reply:
Good direction, thanks

Reply:
done